File: src/main/java/com/refinedmods/refinedstorage/inventory/listener/TileInventoryListener.java
package com.refinedmods.refinedstorage.inventory.listener;

import com.refinedmods.refinedstorage.inventory.item.BaseItemHandler;
import net.minecraft.world.level.block.entity.BlockEntity;

public class TileInventoryListener implements InventoryListener<BaseItemHandler> {
    private final BlockEntity tile;

    public TileInventoryListener(BlockEntity tile) {
        this.tile = tile;
    }

    @Override
    public void onChanged(BaseItemHandler handler, int slot, boolean reading) {
        if (!reading) {
            tile.setChanged();
        }
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/inventory/listener/TileInventoryListener.kt
package com.refinedmods.refinedstorage.inventory.listener

import com.refinedmods.refinedstorage.inventory.item.BaseItemHandler
import net.minecraft.tileentity.BlockEntity

class TileInventoryListener(tile: BlockEntity) : InventoryListener<BaseItemHandler?> {
    private val tile: BlockEntity
    override fun onChanged(handler: BaseItemHandler?, slot: Int, reading: Boolean) {
        if (!reading) {
            tile.markDirty()
        }
    }

    init {
        this.tile = tile
    }
}